### PR TITLE
FSE: Attempt adding SPT tests to CI conf

### PR DESCRIPTION
**Pull request by getdave**
#### Changes proposed in this Pull Request

As per `paAmJe-P4-p2` this attempts to add automated tests runs for the FSE Plugin to the Circle CI config.

### CODE REVIEW

File: .circleci/config.yml
Comment (simison) [inline .circleci/config.yml]: `npm ci` would be little faster and fail if the lockfile is out of sync.

https://docs.npmjs.com/cli/ci.html

File: .circleci/config.yml
Comment (simison) [inline .circleci/config.yml]: Other sections above have `name` defined. Should this one too?

**Review by simison — COMMENTED** [review]
Looking pretty good!
I think you can test this by taking out the draft state and Circle I will run the config. Edit: actually it does look like it's running also as draft?

File: .circleci/config.yml
Comment (getdave) [reply .circleci/config.yml]: Thanks. I wasn't aware of this. I've updated it.

File: .circleci/config.yml
Comment (getdave) [reply .circleci/config.yml]: Good point. Updated.

**Comment by getdave** [comment]
Ok I now have the FSE tests running (and passing!) as part of the CI setup.

Waiting on further reviews before merging.

**Comment by getdave** [comment]
> @automattic/calypso-color-schemes not accessible from @automattic/calypso-build

For some reason I'm now getting a failure with this error.

**Comment by simison** [comment]
I think when your run tests locally, you have root deps installed.

In circle ci you don't I suppose? Not sure what's a good way to go around that if that's the case.

**Comment by sirreal** [comment]
I looked at the current errors in the build, I suspect the problem is that this package doesn't operate as an isolated package, but also isn't an integrated monorepo package.

Based on previous conversations, it's closer to an isolated package, however this line violates that:

https://github.com/Automattic/wp-calypso/blob/6c324d37c52b716ecc240d5f48612c1397512d98/apps/full-site-editing/package.json#L38

This is effectively depending on the "live" calypso-build in the monorepo. It may not work well when depended on by this semi-isolated package.

I'd suggest `npm install -D @automattic/calypso-build` on the FSE package as the shortest way to get this working as expected.

**Comment by getdave** [comment]
Thanks for your advice @sirreal.

> I'd suggest npm install -D @automattic/calypso-build on the FSE package as the shortest way to get this working as expected.

@gwwar How would you feel about this?

**Comment by gwwar** [comment]
> How would you feel about adding @automattic/calypso-build as a explicit dep.

Go for it 👍 I prefer explicit dependencies unless we don't have the time to actually publish a package, or if intertwined dependencies need to be strictly in sync.

**Comment by getdave** [comment]
Looking for an approving 👍 here...

**Review by simison — APPROVED** [review]
Great to have tests run in CI, well done!

We talked a bit about how in-monorepo-but-not-part-of-monorepo situation of FSE might be causing funky dependency deduping issues, especially with `calypso-build` dependency. (p1570560903135700-slack-team-calypso) This might need revisiting later on but in my understanding this is totally fine as of now and safe to merge. 👍

**Pull request merged**
